include/zephyr/sys/util.h

```diff
@@ -783,6 +784,40 @@ static inline void mem_xor_128(uint8_t dst[16], const uint8_t src1[16], const ui
 	mem_xor_n(dst, src1, src2, 16);
 }
 
+/**
+ * @brief Compare memory areas. The same way as `memcmp` it assume areas to be
+ * the same length
+ *
+ * @param m1 First memory area to compare, cannot be NULL even if length is 0
+ * @param m2 Second memory area to compare, cannot be NULL even if length is 0
+ * @param n First n bytes of @p m1 and @p m2 to compares
+ *
+ * @returns true if the @p n first bytes of @p m1 and @p m2 are the same, else
+ * false
+ */
+static inline bool z_util_memeq(const void *m1, const void *m2, size_t n)
```

Comment:
z_ shall not be used for any public APIs, it is a intended to be used for internal/private APIs only.

Reply:
Is that documented anywhere?

The same file has 
https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/sys/util.h#L393-L394
https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/sys/util.h#L731-L745
https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/sys/util.h#L826-L830

Reply:
> Is that documented anywhere?

never made it in the docs, https://github.com/zephyrproject-rtos/zephyr/pull/62198/files. z_ was added long time ago to replace `_` usage in the tree, but unfortuantly it was seen by many as the zephyr designated prefix for APIs and we ended up with public APIs using `z_`....

Reply:
This is something that must be documented. It seems that there is quite a bunch of public APIs now using the `z_`, is there good reasons to continue to have that hidden rule?

To me it seems that without proper documentation some more will definitely be added and then I don't see the point of blocking it here.

Reply:
@nashif do you have alternative suggestions for naming?

Reply:
I updated both names